components/src/core/IconSwitch/IconSwitch.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import { StyleSheet, TouchableOpacity, ViewProps, View } from 'react-native';
+import Animated, {
+    interpolate,
+    useAnimatedStyle,
+    useSharedValue,
+    withTiming,
+    Easing,
+    Extrapolation,
+    interpolateColor,
+} from 'react-native-reanimated';
+import { Icon } from '../Icon/Icon';
+import { $DeepPartial } from '@callstack/react-theme-provider';
+import { MD3Theme, useTheme } from 'react-native-paper';
+
+export type IconSwitchProps = ViewProps & {
+    checkedIcon?: boolean;
+    disabledSwitch?: boolean;
+    color?: string;
+    value?: boolean;
+    onValueChange?: Function;
+    theme?: $DeepPartial<MD3Theme>;
+};
+
+const SWITCH_BUTTON_PADDING = 4;
+const InterpolateXInput = [0, 1];
+
+export const IconSwitch: React.FC<IconSwitchProps> = (props) => {
+    const { checkedIcon = false, disabledSwitch = false, color = 'white', value = false, onValueChange } = props;
+    const theme = useTheme(props.theme);
+
+    const BUTTON_WIDTH = 52;
+    const BUTTON_HEIGHT = 32;
```

Comment:
take these constants and any other styles out of the components, or it will be rendered every cycle.